smk/gene_abundance.smk

```diff
@@ -95,63 +73,40 @@ elif MINTO_MODE in ['reference_genome', 'reference-genome', 'reference', 'refgen
+
+# Alignment filtering
+identity = validate_required_key(config, 'alignment_identity')
+msamtools_filter_length = validate_required_key(config, 'msamtools_filter_length')
 
 mag_omics = 'metaG'
-gene_catalog_path="None"
-gene_catalog_name="None"
+gene_catalog_path = None
+gene_catalog_name = None
+
 if MINTO_MODE == 'MAG':
-    if 'MAG_omics' in config and config['MAG_omics'] != None:
-        mag_omics = config['MAG_omics']
-    reference_dir="{wd}/{mag_omics}/8-1-binning/mags_generation_pipeline/unique_genomes".format(wd = working_dir, mag_omics = mag_omics)
-    print('NOTE: MIntO is using "'+ reference_dir+'" as PATH_reference variable')
+    mag_omics = 'metaG'
+    if (x := validate_optional_key(config, 'MAG_omics')):
+        mag_omics = x
+    reference_dir = f"{working_dir}/{mag_omics}/8-1-binning/mags_generation_pipeline/unique_genomes"
+    print('NOTE: MIntO is using "' + reference_dir + '" as PATH_reference variable')
 else:
-    if config['PATH_reference'] is None:
-        print('ERROR in ', config_path, ': PATH_reference variable is empty. Please, complete ', config_path)
-    elif os.path.exists(config['PATH_reference']) is False:
-        print('ERROR in ', config_path, ': PATH_reference variable path does not exit. Please, complete ', config_path)
-    else:
+    if (x := validate_required_key(config, 'PATH_reference')):
         if MINTO_MODE == 'refgenome':
-            reference_dir=config["PATH_reference"]
+            reference_dir = x
             print('NOTE: MIntO is using "'+ reference_dir+'" as PATH_reference variable')
         elif MINTO_MODE == 'catalog':
-            if os.path.exists(config['PATH_reference']+'/'+config['NAME_reference']) is True:
-                print('NOTE: MIntO is using "'+ config['PATH_reference']+'/'+config['NAME_reference']+'" as PATH_reference and NAME_reference variables.')
-                gene_catalog_path=config["PATH_reference"]
-                gene_catalog_name=config["NAME_reference"]
-            else:
-                print('ERROR in ', config_path, ': NAME_reference variable does not exit. Please, complete ', config_path)
-
-if config['BWA_threads'] is None:
-    print('ERROR in ', config_path, ': BWA_threads variable is empty. Please, complete ', config_path)
-elif type(config['BWA_threads']) != int:
-    print('ERROR in ', config_path, ': BWA_threads variable is not an integer. Please, complete ', config_path)
+            gene_catalog_path = x
+            if (gene_catalog_name := validate_required_key(config, 'NAME_reference')):
+                if os.path.exists(gene_catalog_path + '/' + gene_catalog_name):
+                    print(f"NOTE: MIntO is using {gene_catalog_path}/{gene_catalog_name} as gene database.")
+
+BWA_threads = validate_required_key(config, 'BWA_threads')
 
 if 'MG' in normalization_modes and MINTO_MODE == 'catalog':
     raise Exception("ERROR in {}: In 'catalog' mode, only TPM normalization is allowed.".format(config_path))
```

Comment:
I prefer how the normalization modes are handled in `data_integration.smk`, could that be repeated here as well? You are already checking for it once at line 81.

Reply:
Agreed. Will fix.